services/filemanager/filemanager.go

```diff
@@ -106,6 +110,12 @@ func GetProviderConfigFromEnv() map[string]interface{} {
 		providerConfig["accessKeyID"] = config.GetEnv("AWS_ACCESS_KEY_ID", "")
 		providerConfig["accessKey"] = config.GetEnv("AWS_SECRET_ACCESS_KEY", "")
 		providerConfig["enableSSE"] = config.GetEnvAsBool("AWS_ENABLE_SSE", false)
+		providerConfig["iamRoleArn"] = config.GetEnv("BACKUP_IAM_ROLE_ARN", "")
+		if providerConfig["iamRoleArn"] != "" {
+			// TODO: move this check to startup
+			backendconfig.WaitForConfig(context.TODO())
```

Comment:
Need a better place for this and remove todo.

Reply:
Didn't understand the reason for doing this? 
Why do we try to get workspaceId for empty write key here?`backendconfig.GetWorkspaceIDForWriteKey("")`

Reply:
From what I could infer, the only method we have in the `BackendConfig` interface to get the workspaceId without getting the whole config is `GetWorkspaceIDForWriteKey` with an empty write key (see it being used in similar fashion in couple of other places). In case of hosted/multitenant it would return an empty string and that should be ok as this feature would not anyways apply for those deployments in current format (reading from env instead of something like a org settings set by customer on UI)

Reply:
@chandumlg Can I remove this check completely assuming the failed backups would be retried anyways in case of race condition where backup is triggered before fetching config

Reply:
We are not fetching the config for every attempt. So, retries would fail too because of the old config.

Reply:
@chandumlg doing a `GetWorkspaceIDForWriteKey` here is cool in my opinion. This doesn't seem the place where we should "wait for config". I think the polling with `WaitForConfig()` should be done upstream where we want to bootstrap things (e.g. `main.go` or in the app handlers/dynamic.go, etc...). The BackendConfig will keep on polling on the background even after `WaitForConfig()` returns.

In my head it should work like this:
1. when initializing the service the first time, make sure all is bootstrapped and we got a valid configuration and the polling has started. we wait for a valid config and then say that the service is healthy and proceed with serving requests
2. when needed we can call BackendConfig methods like `GetWorkspaceIDForWriteKey` which will read what we have in memory at the time via a mutex

We have to make sure that we reach `2` only if we were able to start with a valid configuration. Does it make sense?